src/Collision.java

```diff
@@ -0,0 +1,112 @@
+import edu.macalester.graphics.GraphicsGroup;
+import edu.macalester.graphics.Point;
+
+public class Collision {
+
+    private Snake snake;
+
+    private FoodManager foodPieces;
+
+    private WallManager wallManager;
+
+    private GraphicsGroup group;
+
+    
+    /**
+     * 
+     * Tests for various instances for collision throughout the game
+     * 
+     */
+    public Collision (Snake snake, FoodManager foodPieces, WallManager wallManager, GraphicsGroup group) {
+        this.foodPieces = foodPieces;
+        this.snake = snake;
+        this.wallManager = wallManager;
+        this.group = group;
+    }
+    
+    /**
+     * 
+     * Checks for collision between the snake and the food, removing the piece of food if true
+     * 
+     */
+
+    public boolean eatsFood() {
```

Comment:
The class and method naming here is confusing: “a collision eats food??” It might be better to name the class `CollisionDetector` and the method `checkFood`, so that “a collision detector checks food.”